test/e2e/hsts_policy_test.go

```diff
@@ -0,0 +1,157 @@
+		if err := kclient.Delete(context.TODO(), ns); err != nil {
+			t.Fatalf("failed to delete test namespace %s: %v", ns.Name, err)
+		}
+	}()
+
+	// Create pod
+	echoPod := buildEchoPod("hsts-policy-echo", ns.Name)
+	if err := kclient.Create(context.TODO(), echoPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", ns, echoPod.Name, err)
+	}
+
+	// Create service
+	echoService := buildEchoService(echoPod.Name, ns.Name, echoPod.ObjectMeta.Labels)
+	if err := kclient.Create(context.TODO(), echoService); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+	}
+
+	// Create a route that should fail the HSTS policy validation
+	t.Logf("creating an invalid route at %s", time.Now().Format(time.StampMilli))
+	invalidRoute := buildRouteWithHSTS("invalid-route", echoPod.Namespace, echoService.Name, "invalid-route."+appsDomain, "max-age=99999999")
+	if err := kclient.Create(context.TODO(), invalidRoute); err == nil {
+		t.Fatalf("failed to reject an invalid route %s/%s, max-age 99999999", invalidRoute.Namespace, invalidRoute.Name)
+	} else {
+		t.Logf("rejected an invalid route at %s: %s/%s with annotation %s: %v", time.Now().Format(time.StampMilli), invalidRoute.Namespace, invalidRoute.Name, invalidRoute.Annotations, err)
+	}
+
+	// Create a route that should pass the HSTS policy validation
+	exampleHeader := "max-age=0;preload;includesubdomains"
+	t.Logf("creating a valid route at %s", time.Now().Format(time.StampMilli))
+	validRoute := buildRouteWithHSTS("valid-route", echoPod.Namespace, echoService.Name, "valid-route."+appsDomain, exampleHeader)
+	if err := kclient.Create(context.TODO(), validRoute); err != nil {
+		t.Fatalf("failed to create a valid route %s/%s: %v", validRoute.Namespace, validRoute.Name, err)
+	} else {
+		t.Logf("created a valid route at %s: %s/%s with annotation %s", time.Now().Format(time.StampMilli), validRoute.Namespace, validRoute.Name, validRoute.Annotations)
+	}
+
+	code := 0
+	// Wait for route service to respond, and when it does, check for the header that should be there
+	if err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
+		resultHeader, code, _, err := getHttpResponse(t, validRoute)
+		if code != http.StatusOK && err != nil {
+			return false, nil
+		}
+		header := resultHeader.Get("strict-transport-security")
+		if header == "" {
+			t.Logf("request to %s got no HSTS header: %s (%d) %v, trying again....", validRoute.Spec.Host, resultHeader, code, err)
+			return false, nil
+		}
+		if strings.Compare(header, exampleHeader) != 0 {
+			t.Fatalf("request to %s got wrong HSTS header, should be %s, not %s", validRoute.Spec.Host, exampleHeader, header)
+		}
+		t.Logf("request to %s got correct HSTS header: %s", validRoute.Spec.Host, header)
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed to find service %s for header %s in 5 minutes, http code %d: %v", validRoute.Spec.To.Name, exampleHeader, code, err)
```

Comment:
This will always print 0 - see comment https://github.com/openshift/cluster-ingress-operator/pull/639/commits/2ea088ca8f4afe85eb8ebef5b88690856c93202b#r687225592.

Reply:
I'll get rid of that.  I also never return a populated error.